zerver/data_import/mattermost.py

```diff
@@ -0,0 +1,540 @@
+                      user_id_mapper: IdMapper,
+                      user_data_map: Dict[str, Dict[str, Any]],
+                      realm_id: int,
+                      team_name: str) -> None:
+
+    user_data_list = []
+    for username in user_data_map:
+        user = user_data_map[username]
+        if check_user_in_team(user, team_name) or user["is_mirror_dummy"]:
+            user_data_list.append(user)
+
+    for raw_item in user_data_list:
+        user = process_user(raw_item, realm_id, team_name, user_handler, user_id_mapper)
+        user_handler.add_user(user)
+
+def convert_channel_data(channel_data: List[ZerverFieldsT],
+                         user_data_map: Dict[str, Dict[str, Any]],
+                         subscriber_handler: SubscriberHandler,
+                         stream_id_mapper: IdMapper,
+                         user_id_mapper: IdMapper,
+                         realm_id: int,
+                         team_name: str) -> List[ZerverFieldsT]:
+    channel_data_list = [
+        d
+        for d in channel_data
+        if d['team'] == team_name
+    ]
+
+    channel_members_map = {}  # type: Dict[str, List[str]]
+    channel_admins_map = {}  # type: Dict[str, List[str]]
+
+    def initialize_stream_membership_dicts() -> None:
+        for channel in channel_data:
+            channel_name = channel["name"]
+            channel_members_map[channel_name] = []
+            channel_admins_map[channel_name] = []
+
+        for username in user_data_map:
+            user_dict = user_data_map[username]
+            teams = user_dict["teams"]
+            for team in teams:
+                if team["name"] != team_name:
+                    continue
+                for channel in team["channels"]:
+                    channel_roles = channel["roles"]
+                    channel_name = channel["name"]
+                    if "channel_admin" in channel_roles:
+                        channel_admins_map[channel_name].append(username)
+                    elif "channel_user" in channel_roles:
+                        channel_members_map[channel_name].append(username)
+
+    def get_invite_only(v: str) -> bool:
+        if v == 'O':
+            return False
+        elif v == 'P':
+            return True
+        else:
+            raise Exception('unexpected value')
+
+    streams = []
+    initialize_stream_membership_dicts()
+
+    for channel_dict in channel_data_list:
+        now = int(timezone_now().timestamp())
+        stream_id = stream_id_mapper.get(channel_dict['name'])
+        stream_name = channel_dict["name"]
+        invite_only = get_invite_only(channel_dict['type'])
+
+        stream = build_stream(
+            date_created=now,
+            realm_id=realm_id,
+            name=channel_dict['display_name'] or channel_dict['name'],
+            description=channel_dict["purpose"] or channel_dict['header'],
+            stream_id=stream_id,
+            deactivated=False,
```

Comment:
This is something to test with an actual mattermost server -- do they have a notion of a deactivated stream included in exports that we can check for here.

Reply:
They have a similar concept called archived channels. Archived channel data does not appear on the export.json file. They are treated as non-existent. I verified this by creating a channel, doing a export and then archiving the stream and doing another stream. 
![Screenshot from 2019-03-21 18-41-28](https://user-images.githubusercontent.com/7190633/54754245-11e2be80-4c09-11e9-8f13-ce9282d3ca2a.png)


Also the export docs have no mention of archived channels as well.
https://docs.mattermost.com/administration/bulk-export.html#channel-object

Reply:
OK, great, just add a comment to that effect.